qiskit/utils/algorithm_globals.py

```diff
@@ -30,6 +31,14 @@ class QiskitAlgorithmGlobals:
 
     CPU_COUNT = parallel.local_hardware_info()["cpus"]
 
+    @deprecate_func(
+        additional_msg=(
+            "This algorithm utility has been migrated to an independent package: "
+            "https://github.com/qiskit-community/qiskit-algorithms. You can run "
+            "``pip install qiskit_algorithms`` and import ``from qiskit_algorithms.utils`` instead. "
+        ),
+        since="0.45.0",
+    )
     def __init__(self) -> None:
         self._random_seed = None  # type: Optional[int]
         self._num_processes = QiskitAlgorithmGlobals.CPU_COUNT
```

Comment:
This particular part of the deprecation shouldn't ever actually be triggered against users, because users shouldn't be instantiating the class - only the library does that.  If anything, the `algorithms_global` deprecation might have been a little neater having been done using module-level `__getattr__` both here and in `utils/__init__.py`, but let's not change that now.

The reason I mention it is that the current system causes `import qiskit.algorithms` to emit 3 additional warnings - these are correctly blamed on library code so won't be visible to users, but it's indicative of the same situation we had with `opflow` where it turned out that downstream projects testing with all warnings as errors struggled to understand the opflow ones because of excessive errors emitted during the import.